woocommerce/woocommerce-ios
# Order Details: fix shipping address order in different countries

Fixes #1872 

## Why

For certain countries (like some eastern Asian countries), the address starts from the largest unit to the smallest (e.g. from country to the apartment number). Currently, the address in order details just follows the "Region" device setting so all the addresses are formatted the same way (either from the largest to smallest unit or the other way). If the device is set to like Taiwan or Korea, it shows **all** the addresses from the largest to smallest unit. For stores with customers and shipping addresses in different countries, the address might look odd and cause confusion.

This PR sets the country code to `CNPostalAddress` which is used to display the address, so that each address is displayed in the order of the country.

In WC API, an [address's `country` field](https://woocommerce.github.io/woocommerce-rest-api-docs/#order-shipping-properties) follows the ISO 3166-1 alpha-2 format which is the expected value in [`CNPostalAddress`'s `isoCountryCode`](https://developer.apple.com/documentation/contacts/cnpostaladdress/1403396-isocountrycode).

## Testing

- In device settings > General > Language & Region, set the region to Korea or Taiwan
- Launch the app
- Go to the orders tab
- Tap on an order with a physical product (has a shipping address) --> the address should be displayed in the expected order (I think most countries have it from the smallest to largest unit)

## Example screenshots

address country | before | after
-- | -- | --
US | ![Simulator Screen Shot - iPhone 11 - 2021-01-14 at 13 54 26](https://user-images.githubusercontent.com/1945542/104553426-953e9100-5675-11eb-8ab7-a119465b6718.png) | ![Simulator Screen Shot - iPhone 11 - 2021-01-14 at 14 03 54](https://user-images.githubusercontent.com/1945542/104553523-c4550280-5675-11eb-80c9-c05dc264d8de.png)
Taiwan  (technically, the address should only be flipped when it is in Mandarin but I guess iOS doesn't consider the language it is in) | ![Simulator Screen Shot - iPhone 11 - 2021-01-14 at 13 54 44](https://user-images.githubusercontent.com/1945542/104553560-d171f180-5675-11eb-836e-8a46b40e4c8c.png) | ![Simulator Screen Shot - iPhone 11 - 2021-01-14 at 14 04 19](https://user-images.githubusercontent.com/1945542/104553570-d59e0f00-5675-11eb-87b0-3803eba82721.png)

Update release notes:
